src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp

```diff
@@ -0,0 +1,467 @@
+                                          const ProblemDescription& problem,
+                                          const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, problem, invoke_ctx);
+}
+
+bool ConvHipImplicitGemmGroupFwdXdlops::IsApplicable(const ConvolutionContext& ctx,
+                                                     const ProblemDescription& problem) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = problem;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_GROUP_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(problem.conv_problem.GetInDataType() != problem.conv_problem.GetWeightsDataType() ||
+       problem.conv_problem.GetWeightsDataType() != problem.conv_problem.GetOutDataType() ||
+       problem.conv_problem.GetInDataType() != problem.conv_problem.GetOutDataType())
+        return false;
+    if(!problem.direction.IsForward())
+        return false;
+    if(!problem.Is2d())
+        return false;
+    if(!problem.IsLayoutNHWC())
+        return false;
+    const std::string& arch = ctx.GetStream().GetDeviceName();
+    if(!(arch == "gfx908" || arch == "gfx90a"))
+        return false;
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenHalf: return CheckCKApplicability<ck::half_t>(problem);
+    case miopenFloat: return CheckCKApplicability<float>(problem);
+    case miopenInt8: return CheckCKApplicability<int8_t>(problem);
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+ConvSolution ConvHipImplicitGemmGroupFwdXdlops::GetSolution(
+    const ConvolutionContext& ctx,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmGroupFwdXdlops& config) const
+{
+    std::ignore = ctx;
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem;
+    std::ignore = config;
+    return {};
+#else
+    ConvSolution result;
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            switch(problem.conv_problem.GetInDataType())
+            {
+            case miopenHalf:
+                RunCKSolution<ck::half_t>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenFloat:
+                RunCKSolution<float>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenInt8:
+                RunCKSolution<int8_t>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenInt32:
+            case miopenInt8x4:
+            case miopenBFloat16:
+            case miopenDouble: break;
+            }
```

Comment:
[Performance] The same as https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2286#discussion_r1286007620. Computations that depend on problem should be moved from Invoker to InvokerFactory.

Reply:
@iq136boy 

> [Performance] The same as [#2286 (comment)](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2286#discussion_r1286007620). Computations that depend on problem should be moved from Invoker to InvokerFactory.

Original comment updated: added notice that computations can be moved even higher, into `GetSolution`.

Reply:
[resolved]